Pre-processing step for the Python interface

hariszaf opened the pull request:
If applied, this commit will support the pre-processing step for the Python interface
The ```pre_process()``` function is now available and enables this task
The function is located in the ```volestipy.pyx``` however it is purely Python

hariszaf committed:
Create setup_hz.py

hariszaf committed:
Create bindings_hz.cpp

hariszaf committed:
Create volestipy_hz.pyx

hariszaf committed:
Create bindings_hz.h

hariszaf committed:
Update bindings_hz.h

hariszaf committed:
Update bindings_hz.cpp

hariszaf committed:
Create volestipy_old.pyx

hariszaf committed:
Delete volestipy_hz.pyx

hariszaf committed:
Create bindings_old.cpp

hariszaf committed:
Delete bindings_hz.cpp

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Create bindings_old.h

hariszaf committed:
Delete bindings_hz.h

hariszaf committed:
Create setup_old.py

hariszaf committed:
Delete setup_hz.py

hariszaf committed:
Update setup.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.h

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update setup_old.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update setup.py

hariszaf committed:
Update setup.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update test1.py

hariszaf committed:
Create test1_old.py

hariszaf committed:
Update test1.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volume_cb_hpolytope.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Create .gitignore

hariszaf committed:
Delete test1_old.py

hariszaf committed:
Delete setup_old.py

hariszaf committed:
Delete bindings_old.h

hariszaf committed:
Delete bindings_old.cpp

hariszaf committed:
Delete volestipy_old.pyx

hariszaf committed:
merge with the GeomScale develop branch

Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update .gitignore

hariszaf committed:
Update volume_cb_hpolytope.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update test2.py

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update credits.md

hariszaf committed:
on .gitignore: files on test/Testing/Temporary/ were removed from the .gitignore file

hariszaf committed:
first attempt - not working - for rounding

hariszaf committed:
rounding function: does not work; trying to figure it out

hariszaf committed:
round_val now works; still new_A and new_b need to be included

hariszaf committed:
Create metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Create test_rounding.py

hariszaf committed:
Create test1.py

hariszaf committed:
Create test2.py

hariszaf committed:
Delete test2.py

hariszaf committed:
Update test1.py

hariszaf committed:
Create metabolic_net_pipeline-checkpoint.ipynb

hariszaf committed:
Update setup.py

hariszaf committed:
Update .gitignore

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

Last update GeomScale volume_approximation before PR for rounding

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Delete test1.py

git rm test1.py

hariszaf committed:
Delete test2.py

git rm test2.py

hariszaf committed:
Update bindings.h

new class for the preprocessing and the get_full_dimensional_polytope() function in the bindings.h

hariszaf committed:
Update bindings.cpp

new class for the preprocessing and the get_full_dimensional_polytope() function in the bindings.cpp

hariszaf committed:
Update volestipy.pyx

new class for the preprocessing and the get_full_dimensional_polytope() function in the volestipy.pyx

hariszaf committed:
Update bindings.h

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update bindings.cpp

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update volestipy.pyx

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update bindings.h

set member variables for lowDimHPolytopeCPP class

hariszaf committed:
Update bindings.cpp

remove unessecary vars

hariszaf committed:
Update volestipy.pyx

remove unecessary vars in self.full_dim_polytope

hariszaf committed:
Update bindings.h

build the instructor for the full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

1st draft for the get_full_dimensional_polytope() function

hariszaf committed:
Update volestipy.pyx

1st draft for the python interface of the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.h

using transpose matrices for N and A_full
removing the "extra" from the shift and b_full vectors as it is not needed

hariszaf committed:
Update bindings.cpp

2nd draft for the lowDimHPolytopeCPP class.
we changed the N and A_full matrices to transpose to feed them in the cpp file like this.
we removed the _extra from the vectors

hariszaf committed:
Update volestipy.pyx

2nd draft for the lowDimHPolytopeCPP class.
changed "_extra" vectors when needed.
first draft that actually compiles

hariszaf committed:
Create test_get_full_dim_polytope.py

test file for the get_full_dimensional_polytope function
not ready yet

hariszaf committed:
Update test_get_full_dim_polytope.py

changed the Aeq

hariszaf committed:
Update test_get_full_dim_polytope.py

final test for the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

print the N transpose and the full A transpose matrices from the C++ code

hariszaf committed:
Update volestipy.pyx

Attention! What C++ returns is the transpose matrix.
Thus, the [i,j] element of the full A is the [j,i] of what C++ returns. That is what we fixed here

hariszaf committed:
Update test_get_full_dim_polytope.py

print all the outputs of the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

remove test prints

hariszaf committed:
Update volestipy.pyx

remove test printing statements
add all the output in the return statement on the get_full_dim_polytope()
improved comments

hariszaf committed:
Delete metabolic_net_pipeline.ipynb

remove the jupyter notebook

hariszaf committed:
Delete metabolic_net_pipeline-checkpoint.ipynb

remove the hidden directory for the jupyter notebook

hariszaf committed:
Update volestipy.pyx

Exceptions for the low_dim_polytope_cpp constructor added.
Deletion of the A_full and the b_full variables after the initialization of the object.

hariszaf committed:
Update bindings.h

CheBall was removed from the case of the lowDimHPolytopeCPP

hariszaf committed:
Update volestipy.pyx

just some functions of gurobipy

hariszaf committed:
Update volestipy.pyx

first draft for the pre_process() python function

hariszaf committed:
Create test_preprocess.py

a test file for the pre_process() function

hariszaf committed:
Update volestipy.pyx

correct obj function. Aeq line was replaced by a such from A.

hariszaf committed:
Update volestipy.pyx

make dependencies more readable

hariszaf committed:
Update volestipy.pyx

test sparse function

hariszaf committed:
Update test_preprocess.py

change dimensions

hariszaf committed:
Update volestipy.pyx

loop for all the lines of A and models for each of those and its minus

hariszaf committed:
Update test_preprocess.py

change the dimensions

hariszaf committed:
Update volestipy.pyx

draft to check whether it works fine!

hariszaf committed:
Update volestipy.pyx

change name in model.addConstr from "c2" to "c" in all the constraints described, fixed the solutions
remove unecessary print statements

hariszaf committed:
Update test_preprocess.py

add print statement for the results of the pre_process() function

hariszaf committed:
Update test_preprocess.py

build Aeq in a way to be more like to a stoichiometric matrix

hariszaf committed:
Update volestipy.pyx

add model.update() and model.display() function every time we add constraints or change the objective function

hariszaf committed:
Update volestipy.pyx

correct typo in comment

hariszaf committed:
Update volestipy.pyx

add model.update() after the obj function for the minus A[i,]

hariszaf committed:
Update volestipy.pyx

add print statements

hariszaf committed:
Update volestipy.pyx

attempt to solve the issue with the solutions

hariszaf committed:
Update volestipy.pyx

addMConstrs

hariszaf committed:
Update volestipy.pyx

setMObjective()

hariszaf committed:
Update volestipy.pyx

correct objectives()

hariszaf committed:
Update volestipy.pyx

make minus obj function as it should be

hariszaf committed:
Update volestipy.pyx

remove print not needed statements
make code more readable

hariszaf committed:
Update test_preprocess.py

change permission in test

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Merge branch 'develop' into pre_process

hariszaf committed:
Update test_preprocess.py

add import volestipy in test_preprocess.py

TolisChal commented on volestipy/volestipy/volestipy.pyx:
```diff
@@ -225,7 +345,7 @@ cdef class low_dim_HPolytope:
    def dimensions(self):
-      return self._A.shape[1]   
\ No newline at end of file
+      return self._A.shape[1]
```
please add an empty line at the end of all files that contain code in folder `volestipy`

hariszaf replied:
done.

vissarion replied:
these are four lines, one is enough ;)

vissarion replied:
hmm, you are back to the empty line issue now ;)

hariszaf replied:
There is something strange with my editor I think..
If I am not mistaken, last line in the ```volestipy.pyx``` file is number 571 and it is empty..
Is it not?

TolisChal commented on volestipy/volestipy/volestipy.pyx:
```diff
@@ -3,26 +3,146 @@
+         status = model.status
+         if status == GRB.OPTIMAL:
+            solution = model.getAttr('x')
+            print("\n\n***The solution for the MAX case is:")
+            print(solution)
```
Do we need printing here? I think printing should be used only in tests

hariszaf replied:
Nope. I removed the ```solution``` variable completely

TolisChal commented on volestipy/volestipy/volestipy.pyx:
```diff
@@ -3,26 +3,146 @@
+         if status == GRB.OPTIMAL:
+            solution = model.getAttr('x')
+
+            print("\n\n***The solution for the MIN case is:")
+            print(solution)
```
Please remove printing

hariszaf replied:
Likewise

hariszaf committed:
Update volestipy.pyx

remove solutions and their corresponding print statements
add new line in the end of the file

hariszaf committed:
Update volestipy.pyx

correct comment

hariszaf committed:
add bigg file of e.coli in .mat format

hariszaf committed:
add a BIGG file in .json format

hariszaf committed:
add function to read bigg files in .json format; keep the Aeq

hariszaf committed:
develop the read json format bigg file function

hariszaf committed:
add test file for reading json bigg files

hariszaf committed:
add function for reading json bigg files

hariszaf committed:
new directory for the bigg example files

hariszaf committed:
improved function for reading json bigg files; Aeq is now m*n

hariszaf committed:
print metabolites and reaction in the test for json

hariszaf committed:
add function to read .mat bigg files

hariszaf committed:
one test file for reading both json and mat bigg files

hariszaf committed:
remove unecessary print statements

hariszaf committed:
comment in the head of the file describing its purpose

hariszaf committed:
update documentation on how to install volestipy

hariszaf committed:
improved documentation for the dependencies

hariszaf committed:
remember to explain how to get lpsolve

hariszaf committed:
remember to explain how to get lpsolve

hariszaf committed:
remember to explain how to get lpsolve

hariszaf committed:
remember to explain how to get lpsolve

hariszaf committed:
test with RECON1 for the pre-processing step were performed successfully

hariszaf committed:
export .npy files with the outcome of the pre_process function so the user can load them any time

hariszaf committed:
change the input file to e_coli_core

hariszaf committed:
print files with the output of get_full_dim() and rounding() ; start function for inner ball

hariszaf committed:
small changes to check data types

hariszaf committed:
convertions to numpy arrays

hariszaf committed:
complete read-preprocess-get_full_dim steps

vissarion commented on volestipy/README.md:
```diff
@@ -1,28 +1,70 @@
 cd volestipy
-LDFLAGS="-L/usr/lib/lpsolve/" python setup.py build_ext -i
-python setup.py install -f
+LDFLAGS="-L/usr/lib/lp_solve/" python3 setup.py install --user
```
you both use `lp_solve` and 'lpsolve` in README, please be consistent

hariszaf replied:
replcaced ```lp_solve``` with ```lpsolve```

vissarion commented on volestipy/tests/bigg_files/e_coli_core.json:
```diff
@@ -0,0 +1,14786 @@
+{
```
this is extremely huge for inclusion in the library, if it is not publicly available I recommend to create another repository or a dataset (e.g. in zenodo) and put it there

vissarion replied:
EDIT: it is available here: http://bigg.ucsd.edu/models/e_coli_core so you have to retrieve it from there

hariszaf replied:
Files removed from the repo.

vissarion reviewed (commented):
Nice addition, thanks! 

My main comment is about large files (i.e. `json`) and binary (i.e. `.mat`). There should not be in the repo. They are available in the web e.g. http://bigg.ucsd.edu/models/e_coli_core

vissarion commented:
I am trying to run the tests locally. In `volestipy` I am running `python3 tests/test_rounding.py` and get 
```The polytope:
Dimensions: 5
A matrix:
[[ 1.  0.  0.  0.  0.]
 [ 0.  1.  0.  0.  0.]
 [ 0.  0.  1.  0.  0.]
 [ 0.  0.  0.  1.  0.]
 [ 0.  0.  0.  0.  1.]
 [-1.  0.  0.  0.  0.]
 [ 0. -1.  0.  0.  0.]
 [ 0.  0. -1.  0.  0.]
 [ 0.  0.  0. -1.  0.]
 [ 0.  0.  0.  0. -1.]]
<class 'numpy.ndarray'>
b vector:
[ 1.  1.  1.  1.  1.  1.  1.  1.  1.  1.]
<class 'numpy.ndarray'>
*** This is a test for th rounding() function of the volestipy library *** 

Traceback (most recent call last):
  File "tests/test_rounding.py", line 31, in <module>
    rounding_output_max_ellipsoid = hp.rounding(rounding_method = "max_ellipsoid")
  File "volestipy/volestipy.pyx", line 438, in volestipy.HPolytope.rounding
    np.savetxt('round_value.txt', np.asarray(round_value))
  File "/usr/lib/python3/dist-packages/numpy/lib/npyio.py", line 1214, in savetxt
    ncol = X.shape[1]
IndexError: tuple index out of range
```
The rest of the tests seems to run OK.

hariszaf committed:
add data types in .gitignore file

hariszaf committed:
add a test with all the steps of the pipeline

hariszaf committed:
draft for max ball

hariszaf committed:
test for computing max ball

hariszaf committed:
include get_max_ball() function

hariszaf committed:
remove last empty lines

hariszaf committed:
add emppty last line

hariszaf committed:
lp_solve replaced by lpsolve

hariszaf committed:
rounding step included in the full pipeline test

hariszaf committed:
add comput_max_ball in the pipeline

hariszaf committed:
fixed type for the r variable; sre_parse related issue

hariszaf committed:
remove unecessary library

hariszaf committed:
clean up a bit

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

fixed rounding methods

hariszaf committed:
Merge branch 'develop' into pre_process

fixed rounding methods

vissarion reviewed (approved):
Thanks for the revision. If you remove the empty  line issue, I am OK with merging.

hariszaf committed:
Update bindings.cpp

add empty last line

hariszaf committed:
add empty lines in the end of test files

hariszaf committed:
Update test_compute_max_ball.py

add space  in last line

hariszaf committed:
Update test_compute_max_ball.py

hariszaf committed:
Update test_compute_volume.py

hariszaf committed:
Update test_get_full_dim_polytope.py

hariszaf committed:
Update test_random_sampling.py

hariszaf committed:
Update test_read_bigg_files.py

hariszaf committed:
Update test_rounding.py

hariszaf committed:
Update bindings.h
